src/util/address.rs

```diff
@@ -604,7 +612,7 @@ impl Address {
     /// ```
     pub fn is_valid_for_network(&self, network: Network) -> bool {
         let is_legacy = match self.address_type() {
-            Some(AddressType::P2pkh) | Some(AddressType::P2sh) => true,
+            Ok(AddressType::P2pkh) | Ok(AddressType::P2sh) => true,
             _ => false
         };
```

Comment:
Would it be useful to have (public) `is_legacy(&self)` method?

Reply:
How we will define a "legacy"? Just P2PK(H)/P2SH or SegWit v0 as well?

Reply:
Same as this code: not segwit

Although now that you bring it up, perhaps `is_segwit()` would be a more readable name. But more importantly, I'm not sure if this is correct if there's an error: https://github.com/rust-bitcoin/rust-bitcoin/pull/563/files#diff-c55d0186f3d5990b5e223c1a8aa1bbe60988201c452dd13b3324e892bc33119eR624 If it's not then `is_segwit()` instead of `!is_legacy` would seem correct.

Reply:
This code was not introduced by this PR and not related to it, so probably we need to file an issue and do a proper investigation